### PR TITLE
[quant][graphmode][fx] Add functional conv2d + relu

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51155 [quant][graphmode][fx] Add support for functional conv1d and conv3d
* **#51079 [quant][graphmode][fx] Add functional conv2d + relu**

Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26089964](https://our.internmc.facebook.com/intern/diff/D26089964)

### CODE REVIEW

File: torch/quantization/fx/quantization_patterns.py [hunk 1]
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
     def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
                 debug: bool = False,
                 convert_custom_config_dict: Dict[str, Any] = None) -> Node:
+        # Supported combinations are:
+        # quant_type | activation (compute_type) | weight
+        #  static       quint8                      qint8
+        # tuple (activation_dtype, weight_dtype, compute_dtype)
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:224]: is L224 related to L221:L223?  It was a bit confusing to read, I might be missing something

File: torch/quantization/fx/quantization_patterns.py [hunk 2]
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
...
+        supported_dtypes = [
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:225]: is this related to the comments above?

File: torch/quantization/fx/quantization_patterns.py [hunk 3]
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
...
+
         # TODO: debug option for conv module
         qconfig = quantizer.qconfig_map[node.name]
+        dtypes = get_qconfig_dtypes(qconfig)
+        if dtypes not in supported_dtypes:
+            warnings.warn(
+                "dtype combination: {} is not "
+                "supported by Conv "
+                "supported dtype combinations are: {}".format(dtypes, supported_dtypes))
+            if self.relu_node:
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:237]: maybe we can add a comment that this is going to leave the unsupported dtype pattern unquantized? just for easier reading of the code

**Review by vkuzo — APPROVED** [review]

File: torch/quantization/fx/quantization_patterns.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:224]: yeah this is a comment for supported_dtypes

File: torch/quantization/fx/quantization_patterns.py (hunk 2 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:225]: yes

File: torch/quantization/fx/quantization_patterns.py (hunk 3 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:237]: sure
